includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -646,6 +647,39 @@ public function render_post_audio_controls( $content ) {
 			return $content;
 		}
 
+		$audio_timestamp = (int) get_post_meta( $_post->ID, self::AUDIO_TIMESTAMP_KEY, true );
+
+		if ( $audio_timestamp ) {
+			$audio_attachment_url = add_query_arg( 'ver', filter_var( $audio_timestamp, FILTER_SANITIZE_NUMBER_INT ), $audio_attachment_url );
+		}
+
+		/**
+		 * Filters the audio player markup before display.
+		 *
+		 * Returning a non-false value from the filter will short-circuit building
+		 * the block markup and instead will return the custom markup appended to
+		 * the post content.
+		 *
+		 * Note that by using this filter, the custom CSS and JS files will no longer
+		 * be enqueued, so you'll be responsible for either loading them yourself or
+		 * loading custom ones.
+		 *
+		 * @since 2.2.3
+		 *
+		 * @param {bool}     $markup                Audio markup to use. By default false.
+		 * @param {string}   $content               Content of the current post.
+		 * @param {WP_Post}  $_post                 The Post object.
+		 * @param {int}      $audio_attachment_id   The audio attachment ID.
+		 * @param {string}   $audio_attachment_url  The URL to the audio attachment file.
+		 *
+		 * @return {bool|string} Custom audio block markup. Will be appended to the post content.
+		 */
+		$markup = apply_filters( 'classifai_pre_render_post_audio_controls', false, $content, $_post, $audio_attachment_id, $audio_attachment_url );
+
+		if ( false !== $markup ) {
+			return (string) $markup . $content;
```

Comment:
Curious for your thoughts here @joshuaabenazer. The more I think about it, the more I lean towards just returning whatever custom markup someone adds using this new filter, instead of us prepending that to the post content.

For instance, if someone wanted to add a play button to the end of their content, this wouldn't allow them to do that. Since we pass the post content into the filter (as well as the full post object) thinking maybe it's up to whoever uses this filter to ensure they are appending/prepending the post content with their custom markup rather than us hardcoding that here

Reply:
@dkotter Yes, I originally **did not** have the content appended, but I felt like it might make sense after you posted the sample code on Slack. 

It would also depend on how we word the filter too, since it is named`classifai_pre_render_post_audio_controls`, I felt it gives a sense that we are in charge of the audio only. 

Now come to think of it we should just give the control to the developer as that might make it a little less confusing of what is expected. If you agree, then I can work on making that change and possibly change the filter nomenclature too(?) 
I will be able to pick this up early next week though as I sign off early on Fridays.

Also originally I was just going to supply `$content` and `$audio_attachment_url` params to the filter since mostly we would need that only and the remaining can be derived if needed, but it is good to provide the other params since we have it already.

Reply:
> If you agree, then I can work on making that change and possibly change the filter nomenclature too(?)
> I will be able to pick this up early next week though as I sign off early on Fridays.

Yeah, this sounds good to me 👍🏻

Reply:
@dkotter got sidetracked by other priorities. I may be able to get to this on Friday or early next week. That being said I don't mind you taking over to finish this off too.

Reply:
So in looking at this again, I've changed my mind and going to merge this in as-is. Looking at the `render_post_audio_controls` method, it builds the audio markup and prepends that to the post content. So I think having this custom filter do the same thing (take whatever custom markup someone provides and then prepending it to the post content) is fine. If we get complaints or use cases that this doesn't support, happy to look at changing this in the future